structure_tests/drivers/dockerDriver.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drivers
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"context"
+
+	"bufio"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/structure_tests/types/unversioned"
+	"github.com/GoogleCloudPlatform/runtimes-common/structure_tests/utils"
+	"github.com/fsouza/go-dockerclient"
+)
+
+type DockerDriver struct {
+	OriginalImage string
+	CurrentImage  string
+	cli           docker.Client
+}
+
+func (d DockerDriver) Info() string {
+	return fmt.Sprintf("DockerDriver:\nOriginalImage: %s\nCurrentImage: %s\ncli: %s\n",
+		d.OriginalImage, d.CurrentImage, d.cli)
+}
+
+func NewDockerDriver(image string) DockerDriver {
+	newCli, err := docker.NewClientFromEnv()
+	if err != nil {
+		panic(err)
+	}
+	return DockerDriver{
+		OriginalImage: image,
+		CurrentImage:  image,
+		cli:           *newCli,
+	}
+}
+
+func (d DockerDriver) ProcessCommand(t *testing.T, envVars []unversioned.EnvVar, fullCommand []string,
+	shellMode bool, checkOutput bool) (string, string, int) {
+
+	if len(fullCommand) == 0 {
+		t.Logf("empty command provided: skipping...")
+		return "", "", -1
+	}
+	env := []string{}
```

Comment:
nit: `var env []string` is a bit better here, it won't allocate unless you call append.

Reply:
ah yep, good call